storage: fix bug in calcRangeCounter

tbg opened the pull request:
Found by @andreimatei in #39936 (review)

Release note (bug fix): a bug in the range metrics collection could fail
to correctly identify a range that had lost quorum, causing it to not be
reported via the "unavailable ranges" metric. This is now fixed.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45253)
<!-- Reviewable:end -->

tbg commented:
Fixed this up and added (admittedly quite rudimentary) test, PTAL.

andreimatei reviewed (approved):
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45253#-:-M1QwdnPA8dJTdT_S3XC:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @tbg)

___
*[pkg/storage/replica_metrics_test.go, line 37 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45253#-M1QwZUT3_mTYiw8EQrd:-M1QwZUT3_mTYiw8EQre:b5l1uwf) ([raw file](https://github.com/cockroachdb/cockroach/blob/dbf19c6d5a2bc2a5dee882ca3340669bf43b3b57/pkg/storage/replica_metrics_test.go#L37)):*
> ```Go
> 		ctr, down, under, over := calcRangeCounter(1100 /* storeID */, desc, IsLiveMap{
> 			1000: IsLiveMapEntry{IsLive: true}, // by NodeID
> 		}, 4, 3)
> ```

why did you need a 4 here, as opposed to 3?


<!-- Sent from Reviewable.io -->

tbg committed:
storage: fix bug in calcRangeCounter

Found by @andreimatei in https://github.com/cockroachdb/cockroach/pull/39936#pullrequestreview-359059629

Release note (bug fix): a bug in the range metrics collection could fail
to correctly identify a range that had lost quorum, causing it to not be
reported via the "unavailable ranges" metric. This is now fixed.

tbg commented:
I didn't, changed it to 3. TFTR

bors r=andreimatei
